src/test/groovy/com/palantir/gradle/javadist/JavaDistributionPluginTests.groovy

```diff
@@ -61,6 +61,7 @@ class JavaDistributionPluginTests extends GradleTestSpec {
         // try all of the service commands
         exec('dist/service-name-0.1/service/bin/init.sh', 'start') ==~ /(?m)Running 'service-name'\.\.\.\s+Started \(\d+\)\n/
         file('dist/service-name-0.1/var/log/service-name-startup.log', projectDir).text.contains('Test started\n')
+        exec('dist/service-name-0.1/service/bin/init.sh', 'start') ==~ /(?m)Process is already running\n/
```

Comment:
bake it into exec()? yeah that sounds ok.

On Wed, Aug 24, 2016 at 12:15 AM Brad Moylan <notifications@github.com>
wrote:

> In
> src/test/groovy/com/palantir/gradle/javadist/JavaDistributionPluginTests.groovy
> <https://github.com/palantir/gradle-java-distribution/pull/110#discussion_r75992572>
> :
>
> > @@ -61,6 +61,7 @@ class JavaDistributionPluginTests extends GradleTestSpec {
> >          // try all of the service commands
> >          exec('dist/service-name-0.1/service/bin/init.sh', 'start') ==~ /(?m)Running 'service-name'\.\.\.\s+Started \(\d+\)\n/
> >          file('dist/service-name-0.1/var/log/service-name-startup.log', projectDir).text.contains('Test started\n')
> > +        exec('dist/service-name-0.1/service/bin/init.sh', 'start') ==~ /(?m)Process is already running\n/
>
> Yeah I just ran the test on develop and it passed (so doesn't catch the
> bug), working on making exec() check the return code. I think there's an
> assumption throughout this suite that it returns 0 so I can bake in an
> assertion - you cool with that?
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/palantir/gradle-java-distribution/pull/110/files/d60f68511ee7ffd2c83e11735f7a81de586c4c63#r75992572>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGOdweNkSo0licjQCLAaX8qrzeWYCHlBks5qi8VCgaJpZM4Jrmwi>
> .
>